edweiss-firebase/functions/src/utils/custom-sanitizer/quiz.ts

```diff
@@ -65,4 +66,68 @@ export namespace CustomPredicateQuiz {
 			)]
 		})
 
-}
\ No newline at end of file
+	export const isValidLectureQuiz: Predicate<LectureQuizzes.LectureQuiz> =
+
+		Predicate.fields<LectureQuizzes.LectureQuiz>({
+			answer: Predicate.dispatch("type", {
+				MCQAnswersIndices: Predicate.fields({
+					type: Predicate.is("MCQAnswersIndices"),
+					value: Predicate.forEach(Predicate.isNumber)
+				}),
+				TFAnswer: Predicate.fields({
+					type: Predicate.is("TFAnswer"),
+					value: Predicate.isOptionalBoolean
+				}),
+
+
+			}),
+			ended: Predicate.isBoolean,
+			exercise: Predicate.dispatch("type", {
+				MCQ: Predicate.fields({
+					answersIndices: Predicate.forEach(Predicate.isNumber),
+					imageURL: Predicate.isOptionalString,
+					numberOfAnswers: Predicate.isStrictlyPositive,
+					propositions: Predicate.forEach(
+						Predicate.fields<Quizzes.MCQProposition>({
+							description: Predicate.isString,
+							id: Predicate.isPositive,
+							type: Predicate.is("MCQProposition"),
+							imageUrl: Predicate.isOptionalString
+						})
+					),
+					question: Predicate.isString
+				}),
+				TF: Predicate.fields({
+					answer: Predicate.isBoolean,
+					question: Predicate.isString,
+					imageURL: Predicate.isOptionalString
+				})
+			}),
+			showResultToStudents: Predicate.isBoolean
+		});
+
+	export const isValidLectureQuizAttempt: Predicate<LectureQuizzesAttempts.LectureQuizAttempt> =
+
+		Predicate.dispatch("type", {
+			MCQAnswersIndices: Predicate.fields({
+				type: Predicate.is("MCQAnswersIndices"),
+				value: Predicate.forEach(Predicate.isNumber)
+			}),
```

Comment:
I thought about a maximum number of exercises, but I ended up deciding that it should stay indefinite. I could implement a cap of about 100 of course; it just didn't seem all that important. Since we only check with the sanitizer once (when publishing a quiz), I allow this operation.